libdnf5/base/goal.cpp

```diff
@@ -2316,6 +2316,25 @@ base::Transaction Goal::resolve() {
     // TODO(jmracek) Apply modules first
     module::ModuleSack & module_sack = p_impl->base->module_sack;
     ret |= p_impl->add_module_specs_to_goal(transaction);
+    // Check for switched module streams
+    if (!p_impl->base->get_config().get_module_stream_switch_option().get_value()) {
+        auto switched_streams = module_sack.p_impl->module_db->get_all_newly_switched_streams();
+        if (!switched_streams.empty()) {
+            for (auto item : switched_streams) {
+                transaction.p_impl->add_resolve_log(
+                    GoalAction::ENABLE,
+                    GoalProblem::MODULE_CANNOT_SWITH_STREAMS,
+                    GoalJobSettings(),
+                    libdnf5::transaction::TransactionItemType::MODULE,
+                    item.first,
+                    {"0:" + item.second.first, "1:" + item.second.second},
+                    libdnf5::Logger::Level::ERROR);
+            }
+            ret |= GoalProblem::MODULE_CANNOT_SWITH_STREAMS;
+            transaction.p_impl->problems = ret;
+            return transaction;
```

Comment:
There might be a problem because if the operation was called with requesting of a debugsolver data they will be not created.

Reply:
May I ask why you prefer to not continue and collect all problems?

Reply:
The writing of debugsolver data from module solver was not implemented and not called anywhere yet, so this PR doesn't affect that. I believe that this needs to be separate for modules and that it doesn't make sense to continue with resolving the rpm goal if there is a fatal error when resolving modules. But I might be missing something.

Reply:
I think it would be good to create an issue for the debugsolver data for modules.

Reply:
I was thinking about RPM debugsolve code bellow. I completely forget about that modular debugsolv data are not created

Reply:
DNF tries to report all problems at once. If I am not mistaken the problem will suggest to allow module switching by configuration option modification. Think that user will enable module stream switching for the next run and then it will receive another set problems with RPMs.
The second user is API user - even when DNF5 goal.resolve() returns some error then API user can still perform the transaction. But if we will skip some step then the transaction will be incomplete.
Think that user even don't want to run a transaction - they only want to see what happened - what request requires and so on - then it would be good to provide maximum results if it is possible. Switching stream is bad thing, but not a dead end like solver dependency error.

Reply:
Ah, ok, I had the assumption that switching streams is a problem like the solver error, so we should not continue. I'll look into it.